### PR TITLE
chore(types): add String unit tests for Digest and fix NewDigest comment

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

Sunday evening fun after getting sick of seeing horrible digest logged as some weird convoluted fields that don't make sense.

- `NewDigest` comment fixed
- `Test_BlockAnnounceMessage_String` unit test added
- `Test_Digest_String` unit test added

## Tests

```sh
go test -run ^Test_DigestToString$ github.com/ChainSafe/gossamer/dot/types
```

## Issues

One step towards #1945 

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

File: dot/types/digest.go [hunk 1]
```diff
@@ -15,11 +16,40 @@ func NewDigestItem() scale.VaryingDataType {
...
 }
 
-// NewDigest returns a new Digest from the given DigestItems
+// NewDigest returns a new Digest as a varying data type slice.
 func NewDigest() scale.VaryingDataTypeSlice {
 	return scale.NewVaryingDataTypeSlice(NewDigestItem())
 }
 
+// DigestToString renders a digest varying data type slice as a string.
+func DigestToString(digest scale.VaryingDataTypeSlice) (s string) {
```
Comment (EclesioMeloJunior) [inline dot/types/digest.go:25]: is it possible to create a `String` method  at `VaryingDataTypeSlice` struct?

File: dot/types/digest.go [hunk 2]
```diff
@@ -15,11 +16,40 @@ func NewDigestItem() scale.VaryingDataType {
...
+			element = impl.String()
+		case PreRuntimeDigest:
+			element = impl.String()
+		case ConsensusDigest:
+			element = impl.String()
+		case SealDigest:
+			element = impl.String()
+		default:
+			panic(fmt.Sprintf("digest type not implemented for String: %T", impl))
+		}
```
Comment (EclesioMeloJunior) [inline dot/types/digest.go:45]: I think almost all `VaryingDataType` implementations implements `String` method we could change the `VaryingDataTypeValue` interface to:

```go
type VaryingDataTypeValue interface {
	fmt.Stringer
	Index() uint
}
```

Then you can just use `elements = append(elements, vdtValue.String())`

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: dot/types/digest.go (hunk 2 above)
Comment (qdm12) [reply dot/types/digest.go]: Great suggestion. It required a bit more work than expected, a bunch of types were missing the String method, but it's done now in #2970 and `DigestToString` is no longer needed (this PR just ends up being a unit test :wink:)

File: dot/types/digest.go (hunk 1 above)
Comment (qdm12) [reply dot/types/digest.go]: Yes, done in https://github.com/ChainSafe/gossamer/pull/2970 :+1:

**Comment by jimjbrettj** [comment]
Can this be closed now due to https://github.com/ChainSafe/gossamer/pull/2970 ?

**Review by timwu20 — APPROVED** [review]

**Pull request merged**